src/app/credExplorer/pagerankTable/Table.test.js

```diff
@@ -7,50 +7,95 @@ import {NodeAddress, type NodeAddressT} from "../../../core/graph";
 
 import {PagerankTable} from "./Table";
 import {example, COLUMNS} from "./sharedTestUtils";
+import {NodeRowList} from "./Node";
+import {WeightConfig} from "../weights/WeightConfig";
+import {defaultWeightsForAdapter} from "../weights/weights";
+import {FactorioStaticAdapter} from "../../adapters/demoAdapters";
 
 require("../../testUtil").configureEnzyme();
 describe("app/credExplorer/pagerankTable/Table", () => {
   describe("PagerankTable", () => {
-    it("renders thead column order properly", async () => {
-      const {pnd, adapters} = await example();
+    async function setup(defaultNodeFilter?: NodeAddressT) {
+      const {pnd, adapters, weightedTypes} = await example();
+      const onWeightedTypesChange = jest.fn();
+      const maxEntriesPerList = 321;
       const element = shallow(
         <PagerankTable
-          defaultNodeFilter={null}
+          defaultNodeFilter={defaultNodeFilter}
+          weightedTypes={weightedTypes}
+          onWeightedTypesChange={onWeightedTypesChange}
           pnd={pnd}
           adapters={adapters}
-          maxEntriesPerList={1}
+          maxEntriesPerList={maxEntriesPerList}
         />
       );
+      return {pnd, adapters, element, maxEntriesPerList, onWeightedTypesChange};
+    }
+    it("renders thead column order properly", async () => {
+      const {element} = await setup();
       const th = element.find("thead th");
       const columnNames = th.map((t) => t.text());
       expect(columnNames).toEqual(COLUMNS());
     });
 
-    describe("has a filter select", () => {
-      async function setup(defaultNodeFilter?: NodeAddressT) {
-        const {pnd, adapters} = await example();
-        const element = shallow(
-          <PagerankTable
-            defaultNodeFilter={defaultNodeFilter}
-            pnd={pnd}
-            adapters={adapters}
-            maxEntriesPerList={1}
-          />
+    describe("has a WeightConfig", () => {
```

Comment:
Instead of testing against the particular state implementation, why not
test:

  - by default:
      - there is no weight config
      - there is a button that says “Show weight config”
  - when you click the button:
      - there is a weight config
      - the button says “Hide weight config”

Then, the tests are robust to the particular shape of the state field.

Reply:
Ok. Then we also mix in the onWeightedTypeChange test into the "when you click the button" case, which is likely fine.

Reply:
Yep, fine with me, as that is how the user flow works.